crates/hir-ty/src/infer/closure.rs

```diff
@@ -1029,6 +1022,11 @@ impl InferenceContext<'_> {
                 // MIR. I didn't do that due duplicate diagnostics.
                 self.result.mutated_bindings_in_closure.insert(item.place.local);
             }
+
+            // rustc does that only for some captures, not all, because it only uses unique captures
+            // in editions prior to 2021.
+            // We use them in all editions, so we need this here.
```

Comment:
Why do we use them in all editions?

Reply:
Oh, we don't: we don't use `ClosureCapture` at all. I used them in a previous revision of the assist, but this was changed.

It doesn't really matter, but I removed this change.